exporter/pwd_file_test.go

```diff
@@ -0,0 +1,29 @@
+package exporter
+
+import (
+	"log"
+	"os"
+	"testing"
+)
+
+func TestLoadPwdFile(t *testing.T) {
+	if os.Getenv("TEST_REDIS_PWD_FILE") == "" {
+		t.Skipf("TEST_REDIS_PWD_FILE not set - skipping")
+	}
+	passwordMap, err := LoadPwdFile(os.Getenv("TEST_REDIS_PWD_FILE"))
+	if err != nil {
+		t.Fatalf("Test Failed, error: %v", err)
+	}
+
+	if len(passwordMap) == 0 {
+		t.Skipf("Password map is empty -skipping")
+	}
+
+	for host, password := range passwordMap {
```

Comment:
Don't iterate over the map that you load but iterate of a given set of tests where you define the inputs and outputs.

See here for more information and examples: https://github.com/golang/go/wiki/TableDrivenTests